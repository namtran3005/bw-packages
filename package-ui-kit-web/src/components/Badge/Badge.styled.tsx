import * as React from 'react';
import styled from 'styled-components';

import { ButtonBase } from '@material-ui/core';
import { ButtonBaseProps } from '@material-ui/core/ButtonBase';
import { ColorsType, Colors, Spacing } from '@bitwala-cryptobank-squad/package-theme';

import { StyledTypographyProps } from '../../components/Typography/Typography';

interface RootProps extends ButtonBaseProps {
  background?: ColorsType;
  size?: 'normal' | 'small';
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  innerRef?: any;
}

const RootComponent: React.FunctionComponent<RootProps> = ({
  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
  background,
  ...cleanProps
}) => <ButtonBase disableRipple={!cleanProps.onClick} {...cleanProps} />;

const getBadgePadding = (props: RootProps) => {
  if (!props.size || props.size === 'normal') {
    return Spacing.unit / 2;
  } else if (props.size === 'small') {
    return 0;
  }
};

const getBadgeMinDimension = (props: RootProps) => {
  if (!props.size || props.size === 'normal') {
    return Spacing.unit * 3;
  } else if (props.size === 'small') {
    return Spacing.unit * 2;
  }
};

const Root = styled(RootComponent)`
  &&& {
    display: inline-block;
    background-color: ${(props) =>
      Colors.get(props.background || 'primaryGrey')};
    min-width: ${getBadgeMinDimension}px;
    min-height: ${getBadgeMinDimension}px;
    text-align: center;
    border-radius: ${Spacing.unit * Spacing.unit}px;
    padding: 0;
    cursor: ${({ onClick }) => (onClick ? 'pointer' : 'unset')};
    transform: scale(0.75);
    vertical-align: super;
  }
`;

Root.displayName = 'Badge';
RootComponent.displayName = 'BadgeComponent';

export {
  StyledTypographyProps,
  Root,
  RootProps,
  RootComponent,
  getBadgePadding,
};
