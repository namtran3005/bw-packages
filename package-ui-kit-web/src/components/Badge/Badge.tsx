import * as React from 'react';
import { ColorsType } from '@bitwala-cryptobank-squad/package-theme';

import { TypographyProps, Typography } from '../Typography';
import { Root, RootProps } from './Badge.styled';

interface BadgeProps extends RootProps {
  foreground?: ColorsType;
  textAlign?: TypographyProps['textAlign'];
}

const Badge: React.FunctionComponent<BadgeProps> = ({
  children,
  foreground,
  textAlign,
  ...cleanProps
}) => (
  <Root {...cleanProps}>
    <span>
      <Typography.Caption
        textAlign={textAlign || 'center'}
        {...(foreground && { foreground })}
      >
        {children}
      </Typography.Caption>
    </span>
  </Root>
);

export { Badge, BadgeProps };
