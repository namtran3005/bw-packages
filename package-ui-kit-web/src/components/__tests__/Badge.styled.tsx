import * as React from 'react';
import { shallow } from 'enzyme';
import { Root, RootComponent } from '../Badge/Badge.styled';

const defaultProps = {};

describe('Badge', () => {
  it('should render Badge RootComponent', () => {
    expect(shallow(<RootComponent {...defaultProps} />)).toMatchSnapshot();
  });
  it('should render Badge', () => {
    expect(shallow(<Root {...defaultProps} />)).toMatchSnapshot();
  });
  it('should render Badge with background', () => {
    expect(
      shallow(<Root {...defaultProps} background="white" />)
    ).toMatchSnapshot();
  });
});
