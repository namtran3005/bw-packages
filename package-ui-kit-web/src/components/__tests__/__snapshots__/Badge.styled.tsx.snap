// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render Badge 1`] = `
.c0.c0.c0 {
  display: inline-block;
  background-color: #D4D2D2;
  min-width: 24px;
  min-height: 24px;
  text-align: center;
  border-radius: 64px;
  padding: 0;
  cursor: unset;
  -webkit-transform: scale(0.75);
  -ms-transform: scale(0.75);
  transform: scale(0.75);
  vertical-align: super;
}

<BadgeComponent
  className="c0"
/>
`;

exports[`Badge should render Badge RootComponent 1`] = `
<WithStyles(ForwardRef(ButtonBase))
  disableRipple={true}
/>
`;

exports[`Badge should render Badge with background 1`] = `
.c0.c0.c0 {
  display: inline-block;
  background-color: #FFFFFF;
  min-width: 24px;
  min-height: 24px;
  text-align: center;
  border-radius: 64px;
  padding: 0;
  cursor: unset;
  -webkit-transform: scale(0.75);
  -ms-transform: scale(0.75);
  transform: scale(0.75);
  vertical-align: super;
}

<BadgeComponent
  background="white"
  className="c0"
/>
`;
